gcp/observability/go.mod

```diff
@@ -7,7 +7,7 @@ require (
 	github.com/golang/protobuf v1.5.2
 	github.com/google/uuid v1.3.0
 	golang.org/x/oauth2 v0.0.0-20211104180415-d3ed0bb246c8
-	google.golang.org/grpc v1.43.0
+	google.golang.org/grpc v1.46.0
```

Comment:
Did this fail to compile? Since 1.43.0 doesn't have the exported binarylog structs.
Or this was hidden by the `replace` below?

Reply:
Yep... this unfortunately is hidden by the `replace`.

Reply:
So if we make changes in gcp/observability in the future that rely on changes in the main module, we won't ever catch that?  Should we delete the `replace` directive entirely to catch that?  Also does that mean we can't make backward-incompatible changes to our own internal packages now, because a user might be using this version of observability with grpc v1.51.0 one day?  Or do we need to make observability a dependency of the main module and keep them perfectly in sync when we do that?

Reply:
Added a TODO to remove the `replace`.

> does that mean we can't make backward-incompatible changes to our own internal packages now.

It would be great to decouple them (to allow internal packages to continue be internal). But I'm not sure what's the best practice here. We can strive to use let it use public API in future (by exposing experimental APIs).

> Or do we need to make observability a dependency of the main module and keep them perfectly in sync when we do that?

Does this solution mean gRPC Observability will impact the dependency tree of all gRPC Go users? That doesn't sound good...

Reply:
Giving this another thought. We could aim to only use public methods for future plugins like the observability one? In this case, we were using the internal methods of binary logging to customize the logger. We didn't public them, because we were uncertain if we will keep using this design (wrap binary logging) or if the plugin will stay for the long run. We could find a middle ground here to publicize some hooks if they help users to extend those modules.

Reply:
Making the APIs public doesn't make this situation better, because changing them will still break our own observability module.

From another perspective, maybe this isn't that bad, since at least for now, the observability package is still experimental. And also, we don't make separate releases for the observability module (so that even if it's a separate module, it shared the version of the main module).